Open Judge System/OJS.Common/Extensions/FileHelpers.cs

```diff
@@ -82,6 +82,19 @@ public static string FindFileMatchingPattern(string workingDirectory, string pat
             return ProcessModulePath(discoveredFile);
         }
 
+        public static string FindFileMatchingPattern<TOut>(
+            string workingDirectory,
+            string pattern,
+            Func<string, bool> where,
+            Func<string, TOut> orderBy)
+        {
+            var files = DiscoverAllFilesMatchingPattern(workingDirectory, pattern);
+
+            string discoveredFile = files.Where(where).OrderByDescending(orderBy).First();
+
+            return ProcessModulePath(discoveredFile);
+        }
+
         public static void AddFilesToZipArchive(string archivePath, string pathInArchive, params string[] filePaths)
```

Comment:
split row maybe?

Reply:
It is less than 120 characters long (117)